baremetal/metal3machine_manager_test.go

```diff
@@ -597,103 +638,136 @@ var _ = Describe("Metal3Machine manager", func() {
-				M3Machine:        m3mconfig2,
-				ExpectedHostName: host2.Name,
-			}),
 			Entry("Pick hostWithNodeReuseLabelSetToKCP, which has a matching nodeReuseLabelName", testCaseChooseHost{
-				Machine:          newMachine("machine1", "", infrastructureRef2),
-				Hosts:            []client.Object{&hostWithNodeReuseLabelSetToKCP, &host3, &host2},
+				Machine: &clusterv1.Machine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "mymachine",
+						Namespace: namespaceName,
+
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "controlplane.cluster.x-k8s.io/v1beta1",
+								Name:       "test1",
+								Kind:       "KubeadmControlPlane",
+							},
+						},
+						Labels: map[string]string{
+							clusterv1.MachineControlPlaneLabelName: "cluster.x-k8s.io/control-plane",
+						},
+					},
+					Spec: clusterv1.MachineSpec{
+						InfrastructureRef: *infrastructureRef2,
+					},
+				},
+				Hosts:            &bmh.BareMetalHostList{Items: []bmh.BareMetalHost{host1, host2, hostWithNodeReuseLabelSetToKCP}},
 				M3Machine:        m3mconfig2,
 				ExpectedHostName: hostWithNodeReuseLabelSetToKCP.Name,
 			}),
-			Entry("Pick hostWithNodeReuseLabelSetToMD, which has a matching nodeReuseLabelName", testCaseChooseHost{
+			Entry("Pick host2 which lacks ConsumerRef", testCaseChooseHost{
 				Machine:          newMachine("machine1", "", infrastructureRef2),
-				Hosts:            []client.Object{&hostWithNodeReuseLabelSetToMD, &host3, &host2},
+				Hosts:            &bmh.BareMetalHostList{Items: []bmh.BareMetalHost{host1, host2}},
 				M3Machine:        m3mconfig2,
-				ExpectedHostName: hostWithNodeReuseLabelSetToMD.Name,
+				ExpectedHostName: host2.Name,
 			}),
 			Entry("Ignore discoveredHost and pick host2, which lacks a ConsumerRef",
 				testCaseChooseHost{
 					Machine:          newMachine("machine1", "", infrastructureRef2),
-					Hosts:            []client.Object{&discoveredHost, &host2, &host1},
+					Hosts:            &bmh.BareMetalHostList{Items: []bmh.BareMetalHost{discoveredHost, host1, host2}},
 					M3Machine:        m3mconfig2,
 					ExpectedHostName: host2.Name,
 				},
 			),
 			Entry("Ignore hostWithUnhealthyAnnotation and pick host2, which lacks a ConsumerRef",
 				testCaseChooseHost{
 					Machine:          newMachine("machine1", "", infrastructureRef2),
-					Hosts:            []client.Object{&hostWithUnhealthyAnnotation, &host1, &host2},
+					Hosts:            &bmh.BareMetalHostList{Items: []bmh.BareMetalHost{hostWithUnhealthyAnnotation, host1, host2}},
 					M3Machine:        m3mconfig2,
 					ExpectedHostName: host2.Name,
 				},
 			),
 			Entry("Pick host3, which has a matching ConsumerRef", testCaseChooseHost{
-				Machine:          newMachine("machine1", "", infrastructureRef3),
-				Hosts:            []client.Object{&host1, &host3, &host2},
-				M3Machine:        m3mconfig3,
+				Machine:          newMachine("machine1", "", infrastructureRef2),
```

Comment:
this should be:
```suggestion
				Machine:          newMachine("machine1", "", infrastructureRef3),
```
?

Reply:
I fixed this one